token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,216 @@
+        raise e
+
+
+def get_parameters_by_path(path:str, with_decryption=True):
+    """
+    Retrieves parameters from AWS Systems Manager Parameter Store by path.
+
+    Args:
+        path (str): The path to the parameters.
+        with_decryption (bool, optional): Whether to decrypt SecureString parameters. Defaults to True.
+
+    Returns:
+        dict: A dictionary containing the parameter names and values, or None if an error occurs.
+        Returns empty dict if no parameters found on the path.
+    Raises:
+        botocore.exceptions.ClientError: If an error occurs while calling the SSM API.
+    """
+    try:
+        parameters = {}
+        next_token = None
+        while True:
+            if next_token:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption,
+                    NextToken=next_token
+                )
+            else:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption
+                )
+
+            for param in response.get('Parameters', []):
+                parameters[param['Name']] = param['Value']
+
+            next_token = response.get('NextToken')
+            if not next_token:
+                break
+
+        return parameters
+    except(ssm.exceptions.InternalServerError, ssm.exceptions.InvalidFilterKey, ssm.exceptions.InvalidFilterOption,
+           ssm.exceptions.InvalidFilterValue, ssm.exceptions.InvalidKeyId, ssm.exceptions.InvalidNextToken) \
+            as ssm_exception:
+        print(f'ssm get_parameters_by_path error with path: {path} with_decryption:{with_decryption} '
+              f'error: {ssm_exception}')
+        raise ssm_exception
+    except Exception as e:
+        print(f'Unexpected error in get_parameters_by_path with path: {path} with_decryption:{with_decryption} '
+              f'error: {e}')
+        raise e
+
+
+def invoke_lambda(input_params_json, lambda_arn):
+    """
+    Invokes an AWS Lambda function with input parameters from a JSON string.
+
+    Args:
+        input_params_json (str): A JSON string containing input parameters.
+        lambda_arn (str): The ARN of the Lambda function to invoke.
+
+    Returns:
+        str: The JSON-formatted response from the Lambda function, or an error message.
+    """
+    if not lambda_arn:
+        return "Error: lambda_arn not provided"
+
+    try:
+        input_params = json.loads(input_params_json)
+    except json.JSONDecodeError:
+        return f'Error: Invalid JSON : {input_params_json}'
+
+    try:
+        response = lambda_client.invoke(
+            FunctionName=lambda_arn,
+            InvocationType='RequestResponse',  # Synchronous invocation
+            Payload=json.dumps(input_params).encode('utf-8')
+        )
```

Comment:
Why do we deserialize the JSON string just to reserialize it later? Shouldn't we just keep it as a string and assume whatever's calling this passed valid JSON?